scripts/flash/dispute-contribute.js

```diff
@@ -45,76 +44,96 @@ function disputeContribute(augur, args, auth, callback) {
-          }
-          console.log(chalk.green("Market contribute Done"));
-          callback(null);
-        });
-      } else {
-        var marketPayload = { tx: { to: marketId } };
-        augur.api.Market.getFeeWindow(marketPayload, function (err, feeWindowId) {
-          if (err) {
-            console.log(chalk.red(err));
-            return callback("Could not get Fee Window");
-          }
-          if (feeWindowId === "0x0000000000000000000000000000000000000000") {
-            console.log(chalk.red("feeWindowId has not been created"));
-            return callback("Market doesn't have fee window, need to report");
-          }
-          console.log(chalk.yellow("Market Fee Window"), chalk.yellow(feeWindowId));
-          var feeWindowPayload = { tx: { to: feeWindowId } };
-          augur.api.FeeWindow.getStartTime(feeWindowPayload, function (err, feeWindowStartTime) {
+      var universe = augur.contracts.addresses[augur.rpc.getNetworkID()].Universe;
+      getBalance(augur, universe, auth.address, function (err, balances) {
+        if (err) {
+          console.log(chalk.red(err));
+          return callback(JSON.stringify(err));
+        }
+
+        var disputeAmount = amount || balances.reputation;
+        console.log(chalk.yellow.dim("amount"), disputeAmount);
+        var attoREP = speedomatic.fix(disputeAmount, "hex");
+        if (noPush) {
+          console.log(chalk.cyan("Balances:"));
+          console.log("Ether: " + chalk.green(balances.ether));
+          console.log("Rep:   " + chalk.green(balances.reputation));
+          doMarketContribute(augur, marketId, attoREP, payoutNumerators, invalid, auth, function (err) {
+            if (err) {
+              return callback("Market contribute Failed");
+            }
+            console.log(chalk.green("Market contribute Done"));
+            callback(null);
+          });
+
+        } else {
+
+          var marketPayload = { tx: { to: marketId } };
+          augur.api.Market.getFeeWindow(marketPayload, function (err, feeWindowId) {
             if (err) {
               console.log(chalk.red(err));
-              callback("Could not get Fee Window");
+              return callback("Could not get Fee Window");
             }
-            getTime(augur, auth, function (err, timeResult) {
+
+            if (feeWindowId === "0x0000000000000000000000000000000000000000") {
+              console.log(chalk.red("feeWindowId has not been created"));
+              return callback("Market doesn't have fee window, need to report");
+            }
+            console.log(chalk.yellow("Market Fee Window"), chalk.yellow(feeWindowId));
+            var feeWindowPayload = { tx: { to: feeWindowId } };
+            augur.api.FeeWindow.getStartTime(feeWindowPayload, function (err, feeWindowStartTime) {
               if (err) {
                 console.log(chalk.red(err));
-                return callback(err);
+                callback("Could not get Fee Window");
```

Comment:
I think this should be `return callback`

Reply:
good call